### ISSUE TITLE
GUI : Design : Navigation expand and collapse pattern for small screens

### ISSUE THREAD

**Issue by derick-montague** [ibm-openbmc/dev#1646]
## Overview
This has been discussed in the community workgroup. We still need to send out to the community. The design we agreed upon has a hamburger menu that allows the user to expand and collapse the primary navigation.

## Requires discussion
1. Should the hamburger icon be present at all viewport sizes or only on smaller viewport sizes
1. What is breakpoint where the hamburger icon should be visible if the previous is true
1. Does the proposed design work with logos that are taller than what is shown in the proposed layout.
1. Revisit #1292 to determine if/what is needed based on user feeedback

## Proposed Layout
Provided by @ryanarnell based on design by @jandraa

**Comment by derick-montague** [comment]
#1647 is currently in progress. This story will be revisited in regard to customer feedback on #1292.

**Comment by yoshiemuranaka** [comment]
My proposal for responsive layout patterns: 
https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/29333

> 1. Should the hamburger icon be present at all viewport sizes or only on smaller viewport sizes

I suggest hiding the hamburger after reaching a minimum viewport width (Bootstrap default lg bp). I created a `$responsive-layout-bp` variable, so technically if some wanted to **always** have the hamburger menu visible (always be in mobile layout) they could adjust the value of this variable. 

> 2. What is breakpoint where the hamburger icon should be visible if the previous is true

```
The main navigation will be collapsed until the viewport
minimum width reaches the Bootstrap defined 'lg' breakpoint
(defaults to 992px).
```


@ryanarnell @derick-montague 

Let me know what you think or if you want to set aside to look over the code together.

**Comment by derick-montague** [comment]
I think this is a great option. Having that global variable can allow for customization on how / when users are able to show and hide the navigation. If they want it always hidden, they can do that. Downstream, there are many options for customization if needed, but having this one value globally will be a quick and easy way for them to tweak the responsive breakpoint for showing and hiding navigation.

**Comment by derick-montague** [comment]
Closed - Worked in in #1759
